flake.nix

```diff
@@ -0,0 +1,147 @@
+
+  inputs = {
+    flakeCompat.url = github:edolstra/flake-compat;
+    flakeCompat.flake = false;
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = inputs: let
+    commit = inputs.self.shortRev or "dirty";
+    date = inputs.self.lastModifiedDate or inputs.self.lastModified or "19700101";
+    version = "1.2.0+${builtins.substring 0 8 date}.${commit}";
+
+    nixpkgsForHost = host:
+      import inputs.nixpkgs {
+        overlays = [
+          (self: super: {
+            nix-eval-cache = self.rustPlatform.buildRustPackage {
+              pname = "nix-eval-cache";
+              inherit version;
+              src = self.stdenv.mkDerivation {
+                name = "src";
+                builder = builtins.toFile "builder.sh" ''
+                  source $stdenv/setup
+
+                  mkdir $out
+                  cp -rT --no-preserve=mode,ownership $src $out/src/
+                  cp $cargoLock $out/Cargo.lock
+                  cp $cargoToml $out/Cargo.toml
+                '';
+                cargoLock = ./Cargo.lock;
+                cargoToml = ./Cargo.toml;
+                src = ./src;
+              };
+              cargoLock.lockFile = ./Cargo.lock;
+
+              passthru.tests = {
+                version = self.testVersion {package = super.nix-eval-cache;};
+              };
+
+              meta = {
+                description = "The More Aggressive Nix Eval Cache.";
+                homepage = "https://github.com/numtide/nix-eval-cache";
+                license = self.lib.licenses.mit;
+                maintainers = [self.lib.maintainers.mic92];
+                platforms = self.lib.systems.doubles.all;
+              };
+            };
+          })
+        ];
+        system = host;
+      };
+
+    nixpkgs."aarch64-darwin" = nixpkgsForHost "aarch64-darwin";
+    nixpkgs."aarch64-linux" = nixpkgsForHost "aarch64-linux";
+    nixpkgs."i686-linux" = nixpkgsForHost "i686-linux";
+    nixpkgs."x86_64-darwin" = nixpkgsForHost "x86_64-darwin";
+    nixpkgs."x86_64-linux" = nixpkgsForHost "x86_64-linux";
+
+    buildBinariesForHost = host: pkgs: let
+      binaries = builtins.listToAttrs (
+        builtins.map (pkg: {
+          name = "nix-eval-cache-${pkg.stdenv.targetPlatform.config}";
+          value = pkg;
+        })
+        pkgs
+      );
+    in
+      binaries
+      // {
+        "nix-eval-cache-binaries" = nixpkgs.${host}.linkFarm "nix-eval-cache-binaries" (
+          nixpkgs.${host}.lib.mapAttrsToList
+          (name: binary: {
+            inherit name;
+            path = "${binary}/bin/nix-eval-cache";
+          })
+          binaries
+        );
+        "default" = builtins.elemAt pkgs 0;
+      };
+  in rec {
+    checks."aarch64-darwin" = packages."aarch64-darwin";
+    checks."aarch64-linux" = packages."aarch64-linux";
+    checks."i686-linux" = packages."i686-linux";
+    checks."x86_64-darwin" = packages."x86_64-darwin";
```

Comment:
Can you comment out darwin for now? It won't work yet, so I don't want to advertise support for it.